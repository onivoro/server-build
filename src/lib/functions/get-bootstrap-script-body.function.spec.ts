import { getBootstrapScriptBody } from './get-bootstrap-script-body.function';

describe('getBootstrapScriptBody', () => {
    it('templates a script file', () => {
        expect(getBootstrapScriptBody('region-goes-here', 'bucket.goes.here', 'app-goes-here')).toMatchSnapshot();
    });
});