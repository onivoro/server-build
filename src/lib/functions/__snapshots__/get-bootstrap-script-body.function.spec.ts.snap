// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`getBootstrapScriptBody templates a script file 1`] = `
"
    var loadConfig = {
        bucket: 'bucket.goes.here',
        region: 'region-goes-here',
        toLoad: {
            'script': {
                'app-goes-here': ['runtime.js', 'polyfills.js', 'main.js'],
            },
            'link': {
                'app-goes-here': ['styles.css']
            }
        }
    };

    Object.entries(loadConfig.toLoad).forEach((configs) => {
        const type = configs[0];
        const segregatedConfigs = configs[1];
        Object.entries(segregatedConfigs).forEach((segregatedConfig) => {
            const folder = segregatedConfig[0];
            const files = segregatedConfig[1];
            const prefix = \`https://s3.${loadConfig.region}.amazonaws.com/${loadConfig.bucket}\`;

            files.forEach(file => {
                const element = document.createElement(type);
                const path = \`${prefix}/${folder}/${file}\`;

                if (type === 'link') {
                    element.setAttribute('href', path);
                    element.setAttribute('rel', 'stylesheet');
                    document.head.appendChild(element);
                } else if (type === 'script') {
                    element.setAttribute('src', path);
                    element.setAttribute('type', 'module');
                    element.onload = console.warn('onload', path);
                    element.onerror = console.warn('onerror', path);
                    document.head.appendChild(element);
                }
            });
        });
    });

    const l = document.createElement('link');
    l.setAttribute('href', 'https://fonts.googleapis.com/icon?family=Material+Icons');
    l.setAttribute('rel', 'stylesheet');
    document.head.appendChild(l);"
`;
